documentation/swagger-good-patterns.md

```diff
@@ -13,10 +13,19 @@
  - Checks on the Resource Model
     - Every "Resource" Model should have "location" as a "required" property as per the mandate from ARM team; no exceptions are allowed over here.
     - Every "Resource" Model should be tagged with "x-ms-azure-resource". This will cause the Resource to inherit from the Resource definition in the client runtime.
+    - GET/PUT/PATCH for a resource should return the same "Resource" Model, also PUT should accept the same body (there can be readonly, optional, required properties).
+    - For PATCH model every property should be optional and they should not provide any default values.
  - Consistency in the naming of list methods
     - Operations that provide resources in a resource group should be named as "OperationGroup_ListByResourceGroup"
     - Operations that provide resources in a subscription should be named as "OperationGroup_List"
     - If the ListByResourceGroup is overloaded "i.e. if the resourcegroup is null then the operation lists all resources in the subscription" then, such operations should not be allowed.
+ - Consistency of the Names of the APIs, parameters and models
+    - Resource Model name should be same as singular form of the resource type. Like for virtualMachines resource type it should be VirtualMachine.
+    - Parameter names in the apis should be name of the resource like virtualMachines\{virtualMachine} or virtualMachines\{virtualMachineName} both are good example but virtualMachines\{vm} is bad example. 
+    - Any action name should be same as what is defined in the Rest APIs, like regenerateAdminKeys should be regenerateAdminKeys in the Swagger as well, where as regenerateKeys is bad example.
+ - Check for Completeness
+    - Ask for all operations information from the RP, which is usaully can be get from https://management.azure.com/providers/{serviceName}/operations?api-version={apiVersion}.
```

Comment:
Ask for ~~all operations~~ information about all the operations from a RP, which ~~is usually~~ usually can be obtained by making a GET request where the uri is of the following pattern:
- `https://management.azure.com/providers/{serviceName}/operations?api-version={apiVersion}.`